c40509732.lua

```diff
@@ -0,0 +1,88 @@
+--Samurai Destroyer
+function c40509732.initial_effect(c)
+	--synchro summon
+	c:EnableReviveLimit()
+	aux.AddSynchroProcedure(c,nil,aux.NonTuner(),1)
+	--actlimit
+	local e1=Effect.CreateEffect(c)
+	e1:SetType(EFFECT_TYPE_FIELD)
+	e1:SetProperty(EFFECT_FLAG_PLAYER_TARGET)
+	e1:SetCode(EFFECT_CANNOT_ACTIVATE)
+	e1:SetRange(LOCATION_MZONE)
+	e1:SetTargetRange(0,1)
+	e1:SetValue(c40509732.aclimit)
+	e1:SetCondition(c40509732.con)
+	c:RegisterEffect(e1)
+	--disable
+	local e2=Effect.CreateEffect(c)
+	e2:SetType(EFFECT_TYPE_SINGLE+EFFECT_TYPE_CONTINUOUS)
+	e2:SetCode(EVENT_ATTACK_ANNOUNCE)
+	e2:SetRange(LOCATION_MZONE)
+	e2:SetCondition(c40509732.con)
+	e2:SetOperation(c40509732.disop)
+	c:RegisterEffect(e2)
+	local e3=e2:Clone()
+	e3:SetCode(EVENT_BE_BATTLE_TARGET)
+	c:RegisterEffect(e3)
+	--special summon
+	local e4=Effect.CreateEffect(c)
+	e4:SetDescription(aux.Stringid(40509732,0))
+	e4:SetCategory(CATEGORY_SPECIAL_SUMMON)
+	e4:SetType(EFFECT_TYPE_SINGLE+EFFECT_TYPE_TRIGGER_O)
+	e4:SetCode(EVENT_LEAVE_FIELD)
+	e4:SetProperty(EFFECT_FLAG_DELAY+EFFECT_FLAG_CARD_TARGET)
+	e4:SetCondition(c40509732.spcon)
+	e4:SetTarget(c40509732.sptg)
+	e4:SetOperation(c40509732.spop)
+	c:RegisterEffect(e4)
+end
+function c40509732.aclimit(e,re,tp)
+	return not re:GetHandler():IsImmuneToEffect(e)
+end
+function c40509732.con(e)
+	local c=e:GetHandler()
+	return (Duel.GetAttacker()==c and Duel.GetBattleTarget()) or Duel.GetAttackTarget()==c
```

Comment:
there is no Duel.GetBattleTarget
just use `c:GetBattleTarget()` directly

Reply:
https://github.com/Fluorohydride/ygopro-scripts/pull/957/commits/539c4e0a7dac07dfc8a91950dfbb0ea739ae05da fixed.